crates/bevy_ui/src/layout_components.rs

```diff
@@ -0,0 +1,430 @@
+//! Components used to control the layout of UI-[`Node`](super::Node) entities.
+use crate::{Size, UiRect, Val};
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::prelude::Component;
+use bevy_reflect::prelude::*;
+use serde::{Deserialize, Serialize};
+
+use self::flex::FlexLayout;
+
+/// Controls which layout algorithm is used to position this UI node
+#[derive(Component, Copy, Clone, PartialEq, Debug, Serialize, Deserialize, Reflect)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum Layout {
+    /// Use the absolute position and size specified.
+    PositionLayout,
+    /// Use the [Flexbox](https://cssreference.io/flexbox/) layout algorithm.
+    FlexLayout(FlexLayout),
+}
+
+impl Default for Layout {
+    fn default() -> Self {
+        Layout::FlexLayout(FlexLayout::default())
+    }
+}
+
+/// The strategy used to position this node.
+#[derive(
+    Component, Copy, Clone, PartialEq, Eq, Debug, Default, Serialize, Deserialize, Reflect,
+)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum PositionType {
+    /// Relative to all other nodes with the [`PositionType::Relative`] value.
```

Comment:
I don't believe that's correct. If you put a relative position node inside an absolute one, the relative position is still relative to the parent even if the parent is not relative itself. Also it's not even relative to "all the other nodes" but only to the parent.

Reply:
Or more exactly, relative to the position this node would have had as calculated by the layout ignoring this field.